packages/react/src/components/ProgressIndicator/ProgressIndicator.js

```diff
@@ -26,208 +26,219 @@ const defaultTranslations = {
   'carbon.progress-step.invalid': 'Invalid',
 };
 
-function translateWithId(messageId) {
+const translateWithId = (messageId) => {
   return defaultTranslations[messageId];
-}
+};
 
-export function ProgressStep({
-  label,
-  description,
-  className,
-  current,
-  complete,
-  invalid,
-  secondaryLabel,
-  disabled,
-  onClick,
-  renderLabel: ProgressStepLabel,
-  translateWithId: t,
-  ...rest
-}) {
-  const prefix = usePrefix();
-  const classes = classnames({
-    [`${prefix}--progress-step`]: true,
-    [`${prefix}--progress-step--current`]: current,
-    [`${prefix}--progress-step--complete`]: complete,
-    [`${prefix}--progress-step--incomplete`]: !complete && !current,
-    [`${prefix}--progress-step--disabled`]: disabled,
-    [className]: className,
-  });
-
-  const handleKeyDown = (e) => {
-    if (matches(e, [keys.Enter, keys.Space])) {
-      onClick();
-    }
+export class ProgressStep extends React.PureComponent {
+  constructor(props) {
+    super(props);
+  }
```

Comment:
Hey @kristastarr sorry for the late reply. Thank you for that detailed explanation, it's very helpful. I'd really like to help get this into production since it's something our designers are interested in. I think going forward, we can do a few things:

1. Make this an RFC or create a separate RFC to get more input from the rest of our dev team on the implementation. 
2. I am also planning on bringing this up in our next dev team meeting (this week), and hopefully get some thoughts on how we feel about the API changes. Maybe even think about alternate solutions.
3. Based on feedback and input from the rest of the team, we can decide what to do, how to help, etc.

How does that sound?